tensorflow/python/keras/layers/wrappers_test.py

```diff
@@ -290,6 +291,43 @@ def call(self, inputs):
     td3 = keras.layers.TimeDistributed(NoReshapeLayer())
     self.assertFalse(td3._always_use_reshape)
 
+  @tf_test_util.run_in_graph_and_eager_modes
+  def test_TimeDistributed_output_shape_return_types(self):
+
+    class TupleOutputShapeLayer(keras.layers.Layer):
+
+      def call(self, inputs):
+        return inputs
+
+      def compute_output_shape(self, input_shape):
+          return (input_shape[0], input_shape[1])
+
+    class ListOutputShape(keras.layers.Layer):
+
+      def call(self, inputs):
+        return inputs
+
+      def compute_output_shape(self, input_shape):
+          return [input_shape[0], input_shape[1]]
+
+    class TensorShapeOutputShapeLayer(keras.layers.Layer):
+
+      def call(self, inputs):
+        return inputs
+
+      def compute_output_shape(self, input_shape):
+          return tensor_shape.TensorShape([input_shape[0], input_shape[1]])
+
+    test_layers = [TupleOutputShapeLayer, ListOutputShapeLayer,
+                   TensorShapeOutputShapeLayer]
+    # Custom layers can specify output shape as a list, tuple, or TensorShape
+    for layer in test_layers:
+      td = layer()
```

Comment:
What is td? Can we have a more readable variable name here?

Reply:
I originally missed the TimeDistributed wrapper, thanks for catching it.